src/Analysis/Ast/Impl/Caching/CacheFolders.cs

```diff
@@ -103,6 +100,12 @@ internal static class CacheFolders {
                 .Replace('/', '_').Replace('+', '-');
         }
 
+        /// <summary>
+        /// Given a path to a zip file, extracts it to a cache directory and returns the file path to that directory
+        /// </summary>
+        public static string GetZipCacheRoot(IOSPlatform platform)
```

Comment:
I think we should try and make it work without explicit extraction and changing paths. Ex files can be enumerated and read directly via `https://docs.microsoft.com/en-us/dotnet/api/system.io.compression.ziparchiveentry?view=netframework-4.8`.

Reply:
I think it's much easier if we extract them, but I can try and make it work in memory

Reply:
No it is not easier. We'd have to manage the lifetime of extracted files - what if user places new zip with new files? What if there are different versions with the same name but different content? What if we fail to extract all for whatever reasons and now user is stuck with invalid cache and does not know why imports are not getting resolved? What if user has search paths that override what we add or we stomp on their imports by importing something from cache b/c it appeared earlier in search paths? Modifying search paths is pretty risky proposition.